File: catalog/backends/python-django/patterns/audit_log_middleware.py
import logging

# Pattern: SIEM — structured audit log for auth-relevant events
# Adapt: log fields relevant to your domain's security events
audit = logging.getLogger("audit")


class AuditLogMiddleware:
    def __init__(self, get_response):
        self.get_response = get_response

    def __call__(self, request):
        response = self.get_response(request)
        audit.info(
            "request",
            extra={
                "method": request.method,
                "path": request.path,
                "status": response.status_code,
                "remote": request.META.get("REMOTE_ADDR", ""),
            },
        )
        return response

File: catalog/backends/node-express/patterns/audit_log_middleware.js
const pino = require('pino');

// Pattern: SIEM — structured audit log for auth-relevant events
// Adapt: log fields relevant to your domain's security events
const audit = pino({ name: 'audit' });

function auditLogMiddleware(req, res, next) {
  res.on('finish', () => {
    audit.info({
      method: req.method,
      path: req.path,
      status: res.statusCode,
      remote: req.ip,
    }, 'request');
  });
  next();
}

module.exports = { auditLogMiddleware };
